x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_events_query.ts

```diff
@@ -102,17 +126,25 @@ export const buildEventsSearchQuery = ({
   searchAfterSortIds,
   sortOrder,
   timestampOverride,
+  shouldDisableTimestampFallback,
   trackTotalHits,
 }: BuildEventsSearchQuery) => {
   const defaultTimeFields = ['@timestamp'];
   const timestamps =
-    timestampOverride != null ? [timestampOverride, ...defaultTimeFields] : defaultTimeFields;
+    timestampOverride != null
+      ? [timestampOverride, ...(shouldDisableTimestampFallback ? [] : defaultTimeFields)]
+      : defaultTimeFields;
```

Comment:
Similar here, with `primaryTimestamp` and `secondaryTimestamp` we could do `const timestamps = secondaryTimestamp != null ? [primaryTimestamp, secondaryTimestamp] : [primaryTimestamp]` and avoid having the nested conditionals.

Reply:
done